be/src/service/http_service.cpp

```diff
@@ -63,7 +63,9 @@ HttpService::HttpService(ExecEnv* env, int port, int num_threads)
           _ev_http_server(new EvHttpServer(port, num_threads)),
           _web_page_handler(new WebPageHandler(_ev_http_server.get())) {}
 
-HttpService::~HttpService() {}
+HttpService::~HttpService() {
+    stop();
```

Comment:
if the stop method is called inside this class. then it should be a prviate method.